### PR TITLE
[ Rel-5_0 Bug 12482 ] - Fixed showing link of DF in Customer interface

### PR DESCRIPTION
**Pull request by s7design**
https://bugs.otrs.org/show_bug.cgi?id=12482

Hello @mrcbnsls 

This is our proposal for fix this bug. 'a' tag with 'href' link attribute is added in CustomerTicketZoom.tt.
Selenium test is created for checking this.
Please let me know if you have any feedback for this PR.

Regards,
Milan

### CODE REVIEW

**Comment by mgruner** [comment]
Hi Milan. Looking nice! In the agent interface, there is also a 'link preview' feature, which will enable a floater with a preview for a configured URL. Could you add that in the same manner as well? Otherwise we'd wait for the next bug about this. 😁

**Comment by s7design** [comment]
Hi @mgruner 

Thank you for your follow-up. It will be done as soon as possible.

Regards,
Milan

**Comment by s7design** [comment]
Hi @mgruner 

'Link preview' feature is enabled in customer interface as you suggested.
Core.UI.Floater.js is used for floater in the agent interface and also added to customer interface. Usage for interfaces is separated by added IF condition because HTML page structures are not identical for agent and customer interfaces.

Style for floater is added in Customer/Core.Default.css as 'subsection Floaters'. It is similar to agent CSS file.
Settings for left and right floater positions are modified with substraction of 50 pixels to avoid lateral floater edges too close to the left or right side of display.
Link preview inspection is added to Selenium test.

Please let me know if you have any comment for the last commit.

Regards,
Milan

**Comment by s7design** [comment]
Hi @mgruner 

PR is updated with config change.

Regards,
Milan

File: var/httpd/htdocs/js/Core.UI.Floater.js [hunk 1]
```diff
@@ -184,7 +184,13 @@ Core.UI.Floater = (function (TargetNS) {
                 $('[data-trigger="floater"]').removeClass('FloaterOpen');
                 $TriggerObj.addClass('FloaterOpen');
 
-                $FloaterObj = $('#AppWrapper > .MetaFloater.Hidden').clone();
+                // Separate creating of $FloaterObj for agent and customer interface.
+                if (Core.Config.Get('CGIHandle').search(/customer.pl/) > 0) {
```
Comment (mgruner) [inline var/httpd/htdocs/js/Core.UI.Floater.js]: I believe we should consider simplifying that by removing the body/#AppWrapper element from the selector, making this evil switch unneccessary. @mrcbnsls would you agree?

File: var/httpd/htdocs/js/Core.UI.Floater.js [hunk 2]
```diff
@@ -196,15 +202,15 @@ Core.UI.Floater = (function (TargetNS) {
                 FloaterWidth = parseInt($FloaterObj.outerWidth(), 10);
 
                 // set left position
-                if (TriggerOffset.left < FloaterWidth / 2) {
+                if (TriggerOffset.left - 50 < FloaterWidth / 2) {
```
Comment (mgruner) [inline var/httpd/htdocs/js/Core.UI.Floater.js]: @mrcbnsls what do you think about this?

File: Kernel/Output/HTML/Templates/Standard/CustomerTicketZoom.tt [hunk 3]
```diff
@@ -404,9 +404,32 @@
...
+    </a>
+    <i class="fa fa-spinner fa-spin"></i>
+    <div class="NoPreview">
+        <i class="fa fa-warning"></i>
+        [% Translate("A preview of this website can't be provided because it didn't allow to be embedded.") | html %]
+    </div>
+    <div class="Content" style="display: none;">
+        <iframe sandbox="allow-scripts allow-forms"></iframe>
+    </div>
+</div>
```
Comment (mrcbnsls) [inline Kernel/Output/HTML/Templates/Standard/CustomerTicketZoom.tt]: I think this part should go into an include as it is identical to whats included in AgentTicketZoom, right, @s7design?

File: var/httpd/htdocs/js/Core.UI.Floater.js (hunk 2 above)
Comment (mrcbnsls) [inline var/httpd/htdocs/js/Core.UI.Floater.js]: I'm not convinced this is necessary. If possible, things like this should be calculated dynamically from the boundaries of given elements. @s7design, please check if it is possible to calculate it here. And please make sure it's still working in the agent interface, too.

File: Kernel/Output/HTML/Templates/Standard/CustomerTicketZoom.tt [hunk 4]
```diff
@@ -404,9 +404,32 @@
 [% WRAPPER JSOnDocumentComplete %]
 <script type="text/javascript">
     Core.Customer.TicketZoom.Init();
+    if (typeof Core.UI.Floater !== 'undefined') {
+        Core.UI.Floater.Init();
+    }
```
Comment (mrcbnsls) [inline Kernel/Output/HTML/Templates/Standard/CustomerTicketZoom.tt]: Why would Core.UI.Floater possibly be undefined? I don't think this is necessary, is it, @s7design?

**Comment by s7design** [comment]
Hi @mrcbnsls 

PR is updated in accordance with your suggestions.

MetaFloater.tt is created and included to AgentTicketZoom and CustomerTicketZoom. Did you think for this in your comment?

Checking if Core.UI.Floater is undefined is removed.

Calculations for floater classes are changed to dynamic. Other proposal is that this subtraction could be removed.

If you have any comment, please inform me.

Regards,
Milan

**Comment by mrcbnsls** [comment]
Merged successfully (with some small changes) for both rel-5_0 and master. Thanks!

**Pull request closed without merging**